src/sage/rings/polynomial/multi_polynomial_libsingular.pyx

```diff
@@ -5665,7 +5664,7 @@ cdef class MPolynomial_libsingular(MPolynomial_libsingular_base):
             prec = 53
 
         K = FractionField(self.base_ring())
-        if K not in NumberFields() or is_NumberFieldOrder(K):
+        if K not in NumberFields() and not isinstance(K, sage.rings.abc.Order) and K != ZZ:
             raise TypeError("must be over a Numberfield or a Numberfield order")
```

Comment:
Original line 5668 and the modified line 5667 are not logically equivalent. Considering the error message in line 5668, it seems that the original line was wrong. Do you think so?

Reply:
Yes, I think I fixed a bug there.

Reply:
It makes sense. OK.